### PR TITLE
feat: migrate Camera to new arch

### PR DESCRIPTION
**Pull request by WoLewicki**
PR adding Camera implementation on new arch and adding all the files to fabricexample so it is easier to test new things there.

### CODE REVIEW

File: android/src/main/old-arch/com/facebook/react/viewmanagers/RNMBXAndroidTextureMapViewManagerInterface.java [hunk 1]
```diff
@@ -1,37 +1,36 @@
...
 public interface RNMBXAndroidTextureMapViewManagerInterface<T extends View> {
-  void setAttributionEnabled(T view, boolean value);
+  void setAttributionEnabled(T view, Dynamic value);
   void setAttributionPosition(T view, Dynamic value);
-  void setLogoEnabled(T view, boolean value);
+  void setLogoEnabled(T view, Dynamic value);
   void setLogoPosition(T view, Dynamic value);
-  void setCompassEnabled(T view, boolean value);
-  void setCompassFadeWhenNorth(T view, boolean value);
+  void setCompassEnabled(T view, Dynamic value);
```
Comment (mfazekas) [inline android/src/main/old-arch/com/facebook/react/viewmanagers/RNMBXAndroidTextureMapViewManagerInterface.java:20]: What's the reason for the codegen change?  I was changed from `Int`, etc to all Dynamic in `old-arch`?

**Review by mfazekas — COMMENTED** [review]
@WoLewicki nice, thanks much!

I'm don't think I fully understand the need and purpose for `fabricexample`. I'm concerned about code duplication. I'm sure some of the example will be improved in `fabricexample` some in `example` and it will get confusing.

Note that I've updated the example to rn 0.72.5 so it should work with fabric as well, I think I've briefly tried that. It's building in CI for sure.  `fabricexample` still has simpler setup it uses `symlink` instead of metro/babel trickery.

File: android/src/main/old-arch/com/facebook/react/viewmanagers/RNMBXAndroidTextureMapViewManagerInterface.java (hunk 1 above)
Comment (WoLewicki) [reply android/src/main/old-arch/com/facebook/react/viewmanagers/RNMBXAndroidTextureMapViewManagerInterface.java:20]: > I'm don't think I fully understand the need and purpose for fabricexample.

I just wanted to have both example and fabricexample with cached pods etc, but after we convert the whole library, we can just remove it. 

> What's the reason for the codegen change? I was changed from Int, etc to all Dynamic in old-arch?

I'll paste the reasoning I described with folks on codegen channel: 
image

Comment (WoLewicki) [reply android/src/main/old-arch/com/facebook/react/viewmanagers/RNMBXAndroidTextureMapViewManagerInterface.java:20]: I updated the PR with newest main and `UnsafeMixed<T>` code, which is cool btw :D

Comment (mfazekas) [reply android/src/main/old-arch/com/facebook/react/viewmanagers/RNMBXAndroidTextureMapViewManagerInterface.java:20]: Thanks, so just to clarify that I understand correctly. 
```
logoEnabled?: boolean
```

is an optional property and by default we've showed the logo if it was on on the native map, which was true. 
But in new architecture, if user doesn't set's `logoEnabled` logoEnabled gets default value as false, as codegen doesn't recognise the optional `?` in the type declaration.

Have you tested, `WithDefault` 

```
logoEnabled?: WithDefault<boolean,undefined>;
```

for old-arch it seems that it generated correct code:

```
-        mViewManager.setLogoEnabled(view, value == null ? false : (boolean) value);
+        mViewManager.setLogoEnabled(view, value == null ? null : (Boolean) value);
```
Haven't checked what it generated for new arch.

Comment (mfazekas) [reply android/src/main/old-arch/com/facebook/react/viewmanagers/RNMBXAndroidTextureMapViewManagerInterface.java:20]: @WoLewicki oh I see. I assume codegen could (and should) generate `std::optional<boolean>`. But yeah it doesn't. 

Sure than it's good. The only improvement I can think of is this:

```tsx


type OptionalProp<T> = UnsafeMixed<T>;
...

logoEnabled?: OptionalProp<boolean>;

...
}
```

Hopefully codegen will have means of supporting optional properties, and then we can replace.

**Review by mfazekas — APPROVED** [review]
Looks good to me, once the CI pass this can be merged

**Comment by WoLewicki** [comment]
I resolved the issues. You still need to trigger the jobs for CI, right?

**Comment by mfazekas** [comment]
@WoLewicki you have now access I think, so you can push it to a branch on this repo? then CI can run.

**Comment by WoLewicki** [comment]
Ok, I'll check

**Comment by mfazekas** [comment]
@WoLewicki I'd like to update our example to rn 0.73 and will be requiring 0.73 for fabric going further. Is this an issue for you?
0.73 seems to have UnsafeMixed support for events/methods.

**Pull request closed without merging**